### PR TITLE
Add Etcd validation

### PR DESCRIPTION
**Pull request by vanjiii**
**How to categorize this PR?**

/area control-plane
/kind enhancement

**What this PR does / why we need it**:
Include the Etcd validation into the webhook for extension resources.

Update the etcd-druid dependency (this involves applying the breaking changes).

**Which issue(s) this PR fixes**:

**Special notes for your reviewer:**
Currently it uses etcd-druid dependency from `github.com/vanjiii/etcd-druid` fork due to newly introduce logic there. After https://github.com/gardener/etcd-druid/pull/230 is merged this PR will be updated and ready for merging.

### CODE REVIEW

**Review by stoyanr — DISMISSED** [review]
/lgtm
/hold Until the gardener `etcd-druid` (instead of the fork) is vendored.

In case you haven't done it, I would recommend testing an actual reconciliation of a shoot with changed gardenlet code to deliberately generate wrong store prefixes in `Etcd` resources, to check if the validation would actually fail in such cases and prevent the shoot from being reconciled.

**Comment by stoyanr** [comment]
Found two issues while testing this PR:
* The `etcds` resource is missing in the corresponding `validatingwebhookconfiguration`.
* The group / version of the `etcd` resource returned by the `gvr` function in `admission.go` are also wrong.

When I fixed the above, the shoot reconciliation started failing with a wrongly generated Etcd backup store prefix (as expected):
```
"task \"Deploying main and events etcd\" failed: retry failed with
      context deadline exceeded, last error: 1 error occurred:\n\t* admission webhook
      \"validation.extensions.seed.admission.core.gardener.cloud\" denied the request:
      Etcd.druid.gardener.cloud \"shoot--dev--i500152-gcp/etcd-main\" is invalid:
      spec.backup.store.prefix: Invalid value: \"--/etcd-main\": field value must
      contain object name and namespace\n\n"
```

File: pkg/seedadmissioncontroller/webhooks/admission/extensionresources/admission.go
Comment (stoyanr) [inline pkg/seedadmissioncontroller/webhooks/admission/extensionresources/admission.go]: This function will return the wrong resource group and version, you could create e.g. `gvrDruid` to return the correct ones.

**Review by stoyanr — CHANGES_REQUESTED** [review]

**Comment by stoyanr** [comment]
Please also move the line:
```
h.logger.Info("Invalid extension resource detected", "operation", request.Operation, "error", err.Error())
```
before the if statement that currently contains it. It would then be possible to spot validation issues by analysing logs even if `--allow-invalid-extension-resources` is `false`.

File: pkg/seedadmissioncontroller/webhooks/admission/extensionresources/admission.go
Comment (stoyanr) [inline pkg/seedadmissioncontroller/webhooks/admission/extensionresources/admission.go]: You must set both the group and version to `druidv1alpha1.GroupVersion.Group` and `druidv1alpha1.GroupVersion.Version`. Yes, the version happens to be `v1alpha1` in both cases, but this is accidental and should not be relied upon.

Though I have no strong objections, I am not a big fan of the options approach in this case. I think using a function such as `gvrDruid` is simpler and therefore better.

File: example/seed-admission-controller/10-validatingwebhookconfiguration.yaml
Comment (stoyanr) [inline example/seed-admission-controller/10-validatingwebhookconfiguration.yaml:53]: This must be a separate rule with only `etcds` as resource.

**Review by stoyanr — CHANGES_REQUESTED** [review]
Please adjust https://github.com/gardener/gardener/blob/38ad556a7dec689b35c3ee6dd3cf59bd64b5c705/pkg/operation/botanist/component/seedadmissioncontroller/seedadmissioncontroller.go#L399-L419

**Review by stoyanr — COMMENTED** [review]
/lgtm
Can be approved and merged once `gardener/etcd-druid` is properly vendored.

**Review by stoyanr — APPROVED** [review]
/lgtm

From my perspective this could be merged now. Thanks!

**Review by rfranzke — APPROVED** [review]

**Pull request merged**
